Signal/src/ViewControllers/OWSViewController.m

```diff
@@ -110,7 +110,8 @@ - (void)handleKeyboardNotification:(NSNotification *)notification
     CGRect keyboardEndFrameConverted = [self.view convertRect:keyboardEndFrame fromView:nil];
     // Adjust the position of the bottom view to account for the keyboard's
     // intrusion into the view.
-    CGFloat offset = -MAX(0, (self.view.height - keyboardEndFrameConverted.origin.y));
+    // We offset by the bottomLayoutGuide for iPhoneX users, else there is an unnecessary gap between the keyboard and input bar.
+    CGFloat offset = -MAX(0, (self.view.height - keyboardEndFrameConverted.origin.y - self.bottomLayoutGuide.length));
```

Comment:
This is the one I was the least sure of. Take an extra look @charlesmchen.

**Before**
<img width="641" alt="screen shot 2017-11-08 at 9 10 51 am" src="https://user-images.githubusercontent.com/217057/32553554-3990d796-c465-11e7-8eef-cc04a3411993.png">

**After**
<img width="641" alt="screen shot 2017-11-08 at 9 15 21 am" src="https://user-images.githubusercontent.com/217057/32553627-6c9fb210-c465-11e7-8f22-530c420eb504.png">


**Regression Check**
<img width="559" alt="screen shot 2017-11-08 at 9 14 53 am" src="https://user-images.githubusercontent.com/217057/32553586-5023b398-c465-11e7-8d66-64d7a7d12a2e.png">